examples/allocation/zig/README.md

```diff
@@ -0,0 +1,16 @@
+## Zig allocation example
```

Comment:
looks nice. can you update the summary README one level higher, too?

Reply:
Done
https://github.com/tetratelabs/wazero/pull/741/commits/cb48cbec5edfc6bc38876ea6f76f62ee6b5d2c36